frontend/src/services/api.ts

```diff
@@ -34,11 +34,11 @@ export async function fetchReport(): Promise<Report | null> {
   return null;
 }
 
-export async function putLayout(payload: Layout): Promise<Report | null> {
+export async function putView(view: string, layout: Layout): Promise<Project | null> {
   try {
-    const r = await fetch(`${BASE_URL}/report/layout`, {
+    const r = await fetch(`${BASE_URL}/project/views/${view}`, {
```

Comment:
I am not at ease with using names as identifiers in URLs. It often clashes with the `/` character found in the URL path. Could you use a query parameter instead?

For example `/project/views?name=${view}`

Same suggestion for delete and share features.

Reply:
Out of curiosity, what bothers you with this?

Reply:
If you include an identifier in the URL, you want a string over which you have control, that is, something that the program created. Here it is a user-supplied name, hence we have not a lot of control over it, unless we strictly apply validation rules.

Reply:
What about using a proper id?

Reply:
I suggest that we use querystring for now and generalize the usage of id's in another issue.
WDYT ?

Reply:
Query params are now used.